app/models/manageiq/providers/redhat/infra_manager/provision/cloning.rb

```diff
@@ -36,15 +36,33 @@ def prepare_clone_options
 
   def vm_clone_options
     clone_options = {
-      :name       => dest_name,
-      :cluster    => dest_cluster.ems_ref,
-      :clone_type => clone_type,
-      :sparse     => sparse_disk_value
+      :name        => dest_name,
+      :cluster     => dest_cluster.ems_ref,
+      :clone_type  => clone_type,
+      :sparse      => sparse_disk_value,
+      :disk_format => disk_format
     }
     clone_options[:storage] = dest_datastore.ems_ref unless dest_datastore.nil?
     clone_options
   end
 
+  def disk_format
+    # If the datastore is not set, we will go with "block", because this will ensure
+    # the result of the sparsity will be as set by the user.
+    ds_type = "block" unless dest_datastore
+    ds_type ||= dest_datastore.store_type.to_s.upcase == 'iscsi'.upcase ? 'block' : 'file'
```

Comment:
you can replace`dest_datastore.store_type.to_s.upcase == 'iscsi'.upcase` with
dest_datastore.store_type.to_s.downcase == 'iscsi'